core/arch/arm/plat-imx/pm/psci.c

```diff
@@ -61,7 +74,14 @@ int psci_cpu_on(uint32_t core_idx, uint32_t entry,
 		return PSCI_RET_INVALID_PARAMETERS;
 
 	/* set secondary cores' NS entry addresses */
-	ns_entry_addrs[core_idx] = entry;
+	ns_entry_contexts[core_idx].entry_point = entry;
+	ns_entry_contexts[core_idx].r0 = context_id;
+
+	/* flush cache to PoU so other CPUs see the values */
+	cache_op_inner(
+		DCACHE_AREA_CLEAN,
+		&ns_entry_contexts[core_idx],
+		sizeof(struct ns_entry_context));
```

Comment:
minor: review indentation

Is the flush really needed? When the target core will read the values, it is expected to have its secure MMU inited and caches active, no? Did we miss something?

Reply:
No, I don't think the cache flush is needed.